remixer_ui/src/main/java/com/google/android/libraries/remixer/ui/view/RemixerFragment.java

```diff
@@ -79,6 +84,35 @@ public void onClick(View view) {
   }
 
   /**
+  /**
+   * Attach this instance to a shake gesture using {@code sensorManager } and show fragment when magnitude exceeds {@code threshold}
+   */
+  public void attachToShake(final FragmentActivity activity, SensorManager sensorManager, final double threshold) {
+    this.sensorEventListener = new SensorEventListener() {
+      double lastMagnitude;
+      /* Temporary delay until `show` bug is fixed */
+      long lastSpike = 0;
+
+      @Override
+      public void onSensorChanged(SensorEvent event) {
+        double currentMagnitude = Math.sqrt(event.values[0] * event.values[0] + event.values[1] * event.values[1] + event.values[2] * event.values[2]);
+
+        if ((currentMagnitude - lastMagnitude) > threshold && System.currentTimeMillis() - lastSpike > 500) {
+          show(activity.getSupportFragmentManager(), REMIXER_TAG);
+          lastSpike = System.currentTimeMillis();
+        }
+
+        this.lastMagnitude = currentMagnitude;
+      }
+
+      @Override
+      public void onAccuracyChanged(Sensor sensor, int accuracy) { }
+    };
+
+    sensorManager.registerListener(sensorEventListener, sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER), SensorManager.SENSOR_DELAY_NORMAL);
```

Comment:
Experiment with sensor delays and stuff to make sure you're not draining the battery too badly, use the coarsest that will work. 

If you know this is the coarsest that works then ignore this comment.